src/components/PostDetails.tsx

```diff
@@ -1,106 +1,123 @@
-import React from 'react';
+import React, { useContext, useEffect } from 'react';
+import { UserListContext } from './listContext';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { deleteComment } from './todosApi/api-todos';
 
 export const PostDetails: React.FC = () => {
+  const {
+    errorComments,
+    comments,
+    post,
+    selectedPostId,
+    loaderDetails,
+    buttonAddComment,
+    setButtonAddComment,
+    addComment,
+    setAddComent,
+    setButtonLoading,
+    fetchUserComments,
+    setComments,
+  } = useContext(UserListContext);
+
+  const selectedPost = post.find(po => po.id === selectedPostId);
```

Comment:
```suggestion
  const selectedPost = posts.find(post => post.id === selectedPostId);
```
`post` is bad name for array - should be posts